pjsip/src/pjsua-lib/pjsua_acc.c

```diff
@@ -4147,24 +4149,43 @@ pj_status_t pjsua_acc_update_contact_on_ip_change(pjsua_acc *acc)
 	       "by IP change", acc->cfg.id.slen,
 	       acc->cfg.id.ptr, (need_unreg ? "un-" : "")));
 
-    status = pjsua_acc_set_registration(acc->index, !need_unreg);
-
-    if ((status != PJ_SUCCESS) && (pjsua_var.ua_cfg.cb.on_ip_change_progress)
+    status = pjsua_acc_set_registration(acc->index, !need_unreg);    
+    if ((status != PJ_SUCCESS)
 	&& (acc->ip_change_op == PJSUA_IP_CHANGE_OP_ACC_UPDATE_CONTACT))
     {
-	/* If update contact fails, notification might already been triggered
-	 * from registration callback.
-	 */
-	pjsua_ip_change_op_info info;
+	if (status != PJ_EBUSY) {
```

Comment:
I don't understand this either. The title of the PR is to handle if the registration returns busy, but why is the retry done if the status is not busy?